src/ai/ai_help_structs.cc

```diff
@@ -1102,79 +1102,119 @@ uint32_t PlayersStrengths::team_power(Widelands::TeamNumber tn) {
-		if (item.first == pl || item.first == this_player_number || item.second.team_number == 0 ||
-		    item.second.defeated) {
-			continue;
-		}
-		if (can_invite && item.second.team_number == this_player_team) {
-			my_team_sc += item.second.players_diplomacy_score;
-		}
-		if (can_join && item.second.team_number == other_tn) {
-			other_team_sc += item.second.players_diplomacy_score;
+	if (!me_alone || !other_alone) {
+		for (auto& item : all_stats_) {
+			if (item.first == pl || item.first == this_player_number || item.second.team_number == 0 ||
+				 item.second.defeated) {
+				continue;
+			}
+			if (can_invite && item.second.team_number == this_player_team) {
+				my_team_sc += item.second.players_diplomacy_score;
+			}
+			if (can_join && item.second.team_number == other_tn) {
+				other_team_sc += item.second.players_diplomacy_score;
+			}
 		}
 	}
-
 	if (!me_alone) {
 		my_team_sc /= members_in_team(this_player_team) - 1;
 	}
-
 	if (other_alone) {
-		if (other_team_sc - static_cast<int>(RNG::static_rand(10)) <= my_team_sc) {
+		other_team_sc -= RNG::static_rand(10);
+	} else {
+		other_team_sc /= members_in_team(other_tn) - 1;
+	}
+
+	bool invite = false;  // Only to check fall through case
+
+	if (my_team_sc < other_team_sc) {
+		if (other_alone && (me_alone || my_team_sc > static_cast<int>(RNG::static_rand(20)))) {
+			// If we're both alone, we try to be polite and invite.
+			// Or if we have a good enough team, we won't leave it for a lone player, but we try to
+			// recruit.
+			if (can_invite) {
+				invite = true;  // Will fall through
+			} else {
+				verb_log_dbg_time(gametime,
+				                  "AI Diplomacy: Player(%d) with team score (%d) cannot invite player "
+				                  "(%d) with diploscore %d\n",
+				                  static_cast<unsigned int>(this_player_number), my_team_sc,
+				                  static_cast<unsigned int>(pl), get_diplo_score(pl));
+				return;
+			}
+		} else {  // Other has team, or own team is less desirable than other player alone
+			if (!can_join) {
+				const std::string ots_s = other_alone ? "" : format(" and team score %d", other_team_sc);
+				verb_log_dbg_time(gametime,
+				                  "AI Diplomacy: Player(%d) with team score (%d) cannot request to "
+				                  "join player (%d) with diploscore %d%s\n",
+				                  static_cast<unsigned int>(this_player_number), my_team_sc,
+				                  static_cast<unsigned int>(pl), get_diplo_score(pl), ots_s.c_str());
+				return;
+			}
+
+			verb_log_dbg_time(
+			   gametime,
+			   "AI Diplomacy: Player(%d), requests to join player (%d) with diploscore: %d\n",
+			   static_cast<unsigned int>(this_player_number), static_cast<unsigned int>(pl),
+			   get_diplo_score(pl));
+			game.send_player_diplomacy(this_player_number, Widelands::DiplomacyAction::kJoin, pl);
+			set_last_time_requested(gametime, pl);
+			return;
+		}
+	} else {  // Own team is more desirable
+		if (other_alone) {
```

Comment:
why not trying to recruit a lone player with a diploscore of higher then 35 just because the average of my other team mates is even higher.

Reply:
Because it will decrease the team average, especially if we consider that team members are penalized somewhat in the scoring. And bigger teams are not necessarily better, because they can be more unstable.

Reply:
But we may remove the random penalty for lone players, or even make it a small bonus to also invite them if they're close enough. Or make it `score - penalty + random bonus` or even `score - random penalty + random bonus` (even distribution vs. favouring the mean).